### PR TITLE
feat(savings): add Venus Protocol on Network BSC

### PR DESCRIPTION
**Pull request by developerfred**
## Description

add Venus Protocols

Closes # #5790

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

### CODE REVIEW

File: packages/mask/src/plugins/Savings/protocols/VenusProtocol.ts [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+            }
+            return false
+        } catch (error) {
+            return false
+        }
+    }
+
+    public async withdrawEstimate(account: string, chainId: ChainId, web3: Web3, value: BigNumber.Value) {
+        const vTokenAddress = CONTRACT_TOKEN_ADDRESS.find((vault) => vault.address === this.bareToken.address)
+        const TokenAddress = CONTRACT_VBEP_ADDRESS.find((vault) => vault.address === this.stakeToken.address)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Savings/protocols/VenusProtocol.ts:229]:
```suggestion
        const vTokenAddress = CONTRACT_TOKEN_ADDRESS.find((vault) => isSameAddress(vault.address, this.bareToken.address))
        const TokenAddress = CONTRACT_VBEP_ADDRESS.find((vault) => isSameAddress(vault.address, this.stakeToken.address))
```

File: packages/mask/src/plugins/Savings/protocols/VenusProtocol.ts [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+        try {
+            const vbepContract = createContract<Vbep>(web3, this.bareToken.address, vbepABI as AbiItem[])
+
+            const gasEstimate = await vbepContract?.methods
+                .mint(new BigNumber(value).times(new BigNumber(10).pow(this.bareToken.decimals)).toString(10))
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Savings/protocols/VenusProtocol.ts:235]: Use `multipliedBy` and `pow10`.

File: packages/mask/src/plugins/Savings/protocols/VenusProtocol.ts [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+                vbepABI as AbiItem[],
+            )
+            const vToken = createContract<VenusToken>(
+                web3,
+                TokenAddress?.address || ZERO_ADDRESS,
+                venusTokenABI as AbiItem[],
+            )
+
+            vbepContract?.methods
+                .redeemUnderlying(new BigNumber(value).times(new BigNumber(10).pow(18)).integerValue().toString(10))
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Savings/protocols/VenusProtocol.ts:261]: Use `multipliedBy` and `pow10`.

File: packages/mask/src/plugins/Savings/protocols/VenusProtocol.ts [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+                .send({
+                    from: account,
+                    gas: 300000,
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Savings/protocols/VenusProtocol.ts:264]: Avoid dead code gas price.

File: packages/mask/src/plugins/Savings/protocols/VenusProtocol.ts [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+    private async createDepositTokenOperation(account: string, chainId: ChainId, web3: Web3, value: BigNumber.Value) {
+        const vTokenAddress = CONTRACT_TOKEN_ADDRESS.find((vault) => vault.address === this.bareToken.address)
+        const TokenAddress = CONTRACT_VBEP_ADDRESS.find((vault) => vault.address === this.stakeToken.address)
+
+        if (this.bareToken.symbol !== 'bnb') {
+            try {
+                const vbepContract = createContract<Vbep>(web3, this.bareToken.address, vbepABI as AbiItem[])
+
+                await vbepContract?.methods
+                    .mint(new BigNumber(value).times(new BigNumber(10).pow(this.bareToken.decimals)).toString(10))
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Savings/protocols/VenusProtocol.ts:183]: Use `multipliedBy` and `pow10`.

File: packages/mask/src/plugins/Savings/constants/venus.ts [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+    {
+        id: 'busd',
+        symbol: 'BUSD',
+        decimals: 18,
+        address: getTokenConstants(ChainId.BSC).BUSD_ADDRESS,
+    },
+    {
+        id: 'bnb',
+        symbol: 'BNB',
+        decimals: 18,
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Savings/constants/venus.ts:268]: Lost token address here.

**Comment by developerfred** [comment]
@guanbinrui done here 3e6f24f

**Comment by nuanyang233** [comment]
Please remove useless images in `assets`

**Comment by guanbinrui** [comment]
Hi @developerfred Plz resolve those conflicts.

File: packages/mask/src/plugins/Savings/constants/venus.ts [hunk 7]
```diff
@@ -0,0 +1,391 @@
...
+        createERC20Tokens('TRX_ADDRESS', 'TRX', 'TRX', 6)[ChainId.BSC],
+        createERC20Tokens('vTRX_ADDRESS', 'Venus TRX', 'vTRX', vtokenDecimals)[ChainId.BSC],
+    ],
+]
+
+export const CONTRACT_VBEP_ADDRESS = [
+    {
+        id: 'sxp',
+        symbol: 'vSXP',
+        address: getTokenConstants(ChainId.BSC).vSXP_ADDRESS,
```
Comment (septs) [inline packages/mask/src/plugins/Savings/constants/venus.ts:126]: extract the `getTokenConstants(ChainId.BSC)` as variable?

Comment (developerfred) [reply packages/mask/src/plugins/Savings/constants/venus.ts:126]: I'll change it to how we use it in the project.

**Comment by guanbinrui** [comment]
@developerfred We should load these images from a remote URL.

**Comment by developerfred** [comment]
> @developerfred We should load these images from a remote URL.

I'll get it from the token list

**Pull request closed without merging**